matrix.sh

```diff
@@ -54,7 +54,13 @@ help() {
 }
 
 _curl() {
-	curl -s -H "$AUTHORIZATION" -H "User-Agent: matrix.sh/$VERSION" "$@"
+	HEADERFILE="$HOME/.matrix.sh-headers"
```

Comment:
I'd consider using $XDG_CACHE_HOME here, to avoid so many dotfiles in $HOME:

```
HEADERFILE=${XDG_CACHE_HOME:-"$HOME"/.cache}/matrix.sh/headers
mkdir -p "$(dirname "$HEADERFILE")"
```